src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs

```diff
@@ -546,7 +546,7 @@ private Task BindArgumentsAsync()
         {
             var parameterInfo = declaredParameterInfos[index];
 
-            if (!actionParameters.TryGetValue(parameterInfo.Name!, out var value))
+            if (!actionParameters.TryGetValue(parameterInfo.Name!, out var value) || value is null)
```

Comment:
This change feels innocuous enough, but do we know why this is needed? The code around here hasn't changed in ages.

Reply:
The original issue reported the regression between .NET 5 and .NET 6. I was able to verify it as well. A quick peruse of the commit logs has me suspicious about some of the nullability changes that we made in this area for 6.0-preview4 but I, admittedly, haven't dug deeper. 🤷🏽‍♀️